### ISSUE TITLE
Is it possible to install Clarity on a private site? 

### ISSUE THREAD

**Issue by PN4567890** [microsoft/clarity#199]
Hello,

I am new to Clarity and looking to install it on a locally hosted site i.e. no www.domain.com format. Is this possible? If so, I would appreciate direction to the appropriate resources or some instructions on how to do this. 

Thank You

**Comment by ClaritySupport** [comment]
Thanks for your interest in Clarity. Yes Clarity can be installed in locally hosted site.Follow the steps listed[ here](https://docs.microsoft.com/en-us/clarity/getting-started)  to sign up and start using Clarity right away. Refer [Clarity | Microsoft Docs](https://docs.microsoft.com/en-us/clarity/) for more details on Clarity.Please reachout to ClarityMS@microsoft.com for any Clarity related issues.

**Comment by PN4567890** [comment]
Thank you for writing back. That is the method I initially tried but since my site does not have a www.domain.com format, it is being rejected.

**Comment by ClaritySupport** [comment]
The website name doesn’t matter for actual Clarity usage. You can enter name in www.domain.com format and create project then embedded Clarity script from the project in your actual website head section. It will send data to our servers (clarity.ms) and when we try to render those recordings or heatmaps - we won't be able to render those styles or images (so recordings may look broken). CSS/styles must be public in order to render in Clarity.

**Comment by karlholbrook** [comment]
I am not sure the ClaritySupport answer above is correct. We have a private site, the recordings can be made to show the CSS - to a point. If you pause the recording and wait for about one minute the CSS appears and the recording looks good. As soon as you move to another page you get the same problem. Clarity takes at least one minute to perform a redirect (302) to our private CSS. In the F12 tools I can see that its performing a get on
 
https://clarity.microsoft.com/external/resources?url=https://retail.XXXXX.io/Content/css?v=BAYCFExd09HS6c1YzESG0D_Jrt0FMFw6nJ83ITt7E4U1&timestamp=1648212243490

Which results in a redirect (after 43 seconds) to 
https://retail.XXXXX.io/Content/css?v=BAYCFExd09HS6c1YzESG0D_Jrt0FMFw6nJ83ITt7E4U1

The CSS then loads successfully. 
![image](https://user-images.githubusercontent.com/48241719/160405476-94cc8efb-67e4-47c6-9429-23c361951b79.png)

**Comment by karlholbrook** [comment]
Hi, yes
https://clarity.microsoft.com/player/azcpe03kvi/um41y8/eeozq3/1
If I pause the recording at about 4 seconds in then wait 30 seconds the CSS shows.

**Comment by ClaritySupport** [comment]
Are your resources publicly accessible? we tried loading the them on Postman and on a browser, and the requests timeout. For Clarity to render the recordings/heatmaps correctly, the styles used must be available and publicly accessible with no auth/intranet/any special requirements .

**Comment by ClaritySupport** [comment]
Resource is private which is why we cannot fetch it and it fails. When it fails it redirects the browser to fetch them then the browser might have used some cookies or intranet to successfully find it.

**Comment by karlholbrook** [comment]
Yes that is exactly what we are seeing. The redirect takes about 40 seconds to realise it can't find the resource. As soon as the redirect completes the resource is loaded from the browser (intranet) very quickly. Is there a way to stop the redirect to clarity and look on our intranet first, or at least give up on the redirect quickly?

**Comment by ClaritySupport** [comment]
@karlholbrook , Are you still experiencing this issue? if yes, please do share link to recordings/Heatmaps to investigate further.
